src/pytest_workflow/plugin.py

```diff
@@ -443,20 +443,21 @@ def collect(self):
         # collection phase.
         workflow = self.queue_workflow()
 
+        workflow.desired_exit_code = self.workflow_test.exit_code
```

Comment:
It is better to do this in the self.queue_workflow function so that the desired exit code can be set upon class instantion rather than bolted on later.